### PR TITLE
gui/firmware and game install: add check for delete file after installed.

### PR DESCRIPTION
**Pull request by Zangetsu38**
- Add checkbox for popup dialog in firmware install and game install.
- Add firmware version on pop confirmation after install fw.

### CODE REVIEW

File: vita3k/gui/src/game_install_dialog.cpp [hunk 1]
```diff
@@ -56,6 +56,7 @@ void draw_game_install_dialog(GuiState &gui, HostState &host) {
     }
 
     static const auto BUTTON_SIZE = ImVec2(60.f, 0.f);
+    static bool delete_game_file = false;
```
Comment (1whatleytay) [inline vita3k/gui/src/game_install_dialog.cpp]: Please don't make these static inside a method. Make them global. I beg of you!

File: vita3k/gui/src/game_install_dialog.cpp [hunk 2]
```diff
@@ -93,9 +94,14 @@ void draw_game_install_dialog(GuiState &gui, HostState &host) {
         ImGui::PopStyleColor();
         ImGui::TextColored(GUI_COLOR_TEXT, "Succesfully installed game.\n%s [%s] %s", host.io.title_id.c_str(), host.game_title.c_str(), host.game_version.c_str());
         ImGui::Separator();
+        ImGui::Checkbox("Delete Game file installer?", &delete_game_file);
```
Comment (1whatleytay) [inline vita3k/gui/src/game_install_dialog.cpp]: Hmm... I'm not sure what you mean here...

Did you mean `Delete vpk after installation?`

File: vita3k/gui/src/game_install_dialog.cpp [hunk 3]
```diff
@@ -110,10 +116,15 @@ void draw_game_install_dialog(GuiState &gui, HostState &host) {
     if (ImGui::BeginPopupModal("Game installation failed", nullptr, ImGuiWindowFlags_AlwaysAutoResize)) {
         ImGui::PopStyleColor();
         ImGui::TextColored(GUI_COLOR_TEXT, "Installation failed, please check the log for more information.");
+        ImGui::Checkbox("Delete Game file installer?", &delete_game_file);
```
Comment (1whatleytay) [inline vita3k/gui/src/game_install_dialog.cpp]: Same message in two places in the same method? 🤔 

Just curious. No change requested.

Comment (Zangetsu38) [reply vita3k/gui/src/game_install_dialog.cpp]: Is for install succès and install failed so need write same 2 time

File: vita3k/gui/src/game_install_dialog.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/gui/src/game_install_dialog.cpp]: No i keep it here like other is already set

Comment (1whatleytay) [reply vita3k/gui/src/game_install_dialog.cpp]: It's not hard to move it to global, it's just a copy paste...

Please...

Comment (VelocityRa) [inline vita3k/gui/src/game_install_dialog.cpp]: stop using `static` for non-const variables
make this part of `GuiState` or something

File: vita3k/gui/src/firmware_install_dialog.cpp [hunk 4]
```diff
@@ -28,8 +28,24 @@
 
 namespace gui {
 
+static std::string fw_version = {};
+
+static void get_firmware_version (HostState &host) {
```
Comment (VelocityRa) [inline vita3k/gui/src/firmware_install_dialog.cpp]: wrong style, run format script

File: vita3k/gui/src/firmware_install_dialog.cpp [hunk 5]
```diff
@@ -28,8 +28,24 @@
 
 namespace gui {
 
+static std::string fw_version = {};
```
Comment (VelocityRa) [inline vita3k/gui/src/firmware_install_dialog.cpp]: ditto, don't use `static`

**Comment by Zangetsu38** [comment]
change request is done

**Comment by Zangetsu38** [comment]
clean both code, totally forget remove include set before for test

File: vita3k/gui/src/firmware_install_dialog.cpp [hunk 6]
```diff
@@ -15,21 +15,33 @@
...
+static void get_firmware_version(HostState &host) {
+    std::ifstream versionFile(host.pref_path + "/PUP_DEC/PUP/version.txt");
+
+    if (versionFile.is_open()) {
+        std::getline(versionFile, fw_version);
+        versionFile.close();
+    } else
+        LOG_WARN("Firmware Version file not found!");
+
+    fs::remove_all(host.pref_path + "/PUP_DEC");
```
Comment (1whatleytay) [inline vita3k/gui/src/firmware_install_dialog.cpp]:
```suggestion
    fs::remove_all(fs::path(host.pref_path) / "PUP_DEC");
```
Above is less likely to have duplicate slashes I think.

You could do the same for a few lines above.

File: vita3k/gui/src/firmware_install_dialog.cpp [hunk 7]
```diff
@@ -54,9 +67,22 @@ void draw_firmware_install_dialog(GuiState &gui, HostState &host) {
...
+            ImGui::TextColored(GUI_COLOR_TEXT, "Firmware version: %s", fw_version.c_str());
+        ImGui::Spacing();
+        ImGui::Separator();
+        ImGui::Spacing();
+        ImGui::Checkbox("Delete the firmware installation file?", &delete_fw_file);
+        ImGui::Spacing();
         ImGui::SetCursorPosX(ImGui::GetWindowWidth() / 2 - 30);
         if (ImGui::Button("OK", BUTTON_SIZE)) {
+            if (delete_fw_file) {
+                fs::remove(static_cast<const fs::path>(fw_path));
```
Comment (1whatleytay) [inline vita3k/gui/src/firmware_install_dialog.cpp]:
```suggestion
                fs::remove(fs::path(fw_path));
```
This is a bit shorter and the cast is probably calling the constructor anyway.

File: vita3k/gui/src/game_install_dialog.cpp [hunk 8]
```diff
@@ -91,11 +84,19 @@ void draw_game_install_dialog(GuiState &gui, HostState &host) {
     ImGui::PushStyleColor(ImGuiCol_Text, GUI_COLOR_TEXT_MENUBAR_OPTIONS);
     if (ImGui::BeginPopupModal("Game installation succes", &game_install_confirm, ImGuiWindowFlags_AlwaysAutoResize)) {
         ImGui::PopStyleColor();
+        ImGui::SetCursorPosX(ImGui::GetWindowWidth() / 5);
         ImGui::TextColored(GUI_COLOR_TEXT, "Succesfully installed game.\n%s [%s] %s", host.io.title_id.c_str(), host.game_title.c_str(), host.game_version.c_str());
+        ImGui::Spacing();
         ImGui::Separator();
         ImGui::Spacing();
+        ImGui::Checkbox("Delete the package from the installed app?", &delete_game_file);
```
Comment (1whatleytay) [inline vita3k/gui/src/game_install_dialog.cpp]:
```suggestion
        ImGui::Checkbox("Delete the package for the installed app?", &delete_game_file);
```

File: vita3k/gui/src/game_install_dialog.cpp [hunk 9]
```diff
@@ -91,11 +84,19 @@ void draw_game_install_dialog(GuiState &gui, HostState &host) {
...
+        ImGui::Spacing();
         ImGui::SetCursorPosX(ImGui::GetWindowWidth() / 2 - 30);
         if (ImGui::Button("Ok", BUTTON_SIZE)) {
+            if (delete_game_file) {
+                fs::remove(static_cast<const fs::path>(game_path));
```
Comment (1whatleytay) [inline vita3k/gui/src/game_install_dialog.cpp]:
```suggestion
                fs::remove(game_path);
```
Does it not let you do this?

If not:
```suggestion
                fs::remove(fs::path(game_path));
```

File: vita3k/gui/src/game_install_dialog.cpp [hunk 10]
```diff
@@ -110,10 +111,17 @@ void draw_game_install_dialog(GuiState &gui, HostState &host) {
...
         ImGui::TextColored(GUI_COLOR_TEXT, "Installation failed, please check the log for more information.");
+        ImGui::Spacing();
         ImGui::Separator();
         ImGui::Spacing();
+        ImGui::Checkbox("Delete the package from the app that failed to install?", &delete_game_file);
+        ImGui::Spacing();
         ImGui::SetCursorPosX(ImGui::GetWindowWidth() / 2 - 30);
         if (ImGui::Button("Ok", BUTTON_SIZE)) {
+            if (delete_game_file) {
+                fs::remove(static_cast<const fs::path>(game_path));
```
Comment (1whatleytay) [inline vita3k/gui/src/game_install_dialog.cpp]: Ditto.

File: vita3k/gui/src/firmware_install_dialog.cpp [hunk 11]
```diff
@@ -15,21 +15,33 @@
...
 #include <nfd.h>
 
 namespace gui {
 
+std::string fw_version;
+bool delete_fw_file = false;
+nfdchar_t *fw_path = nullptr;
+
+static void get_firmware_version(HostState &host) {
+    std::ifstream versionFile(host.pref_path + "/PUP_DEC/PUP/version.txt");
```
Comment (1whatleytay) [inline vita3k/gui/src/firmware_install_dialog.cpp]: you could also do it here but then you'd have to convert it back :P

File: vita3k/gui/src/game_install_dialog.cpp [hunk 12]
```diff
@@ -19,22 +19,15 @@
...
-
-#include <io/vfs.h>
-
-#include <util/log.h>
 
 #include <nfd.h>
 
 namespace gui {
 
+bool delete_game_file = false;
```
Comment (VelocityRa) [inline vita3k/gui/src/game_install_dialog.cpp]: so you just removed the `static` keyword? lol
the issue is not with just static
don't use globals
put everything in `GuiState` like I told you

Comment (Zangetsu38) [reply vita3k/gui/src/game_install_dialog.cpp]: I can't here for one things, and master already use global few time for code, and here is only used for this file, not need est this on gui state

Comment (VelocityRa) [reply vita3k/gui/src/game_install_dialog.cpp]: Why can't you?
We use globals when it's necessary, not out of lazyness.

Comment (Zangetsu38) [reply vita3k/gui/src/game_install_dialog.cpp]: i can't like ni said, is not possible for nfd

Comment (VelocityRa) [reply vita3k/gui/src/game_install_dialog.cpp]: Do it for the other 3 globals then...

Comment (Zangetsu38) [reply vita3k/gui/src/game_install_dialog.cpp]: if i keep one here, i can also keep other, is not need set it on gui state, only this file using it but here i try somethings

Comment (VelocityRa) [reply vita3k/gui/src/game_install_dialog.cpp]: Leave it, I'll just merge.

**Pull request merged**